tuned/plugins/base.py

```diff
@@ -298,10 +298,13 @@ def instance_update_tuning(self, instance):
 		if instance.has_dynamic_tuning and self._global_cfg.get(consts.CFG_DYNAMIC_TUNING, consts.CFG_DEF_DYNAMIC_TUNING):
 			self._run_for_each_device(instance, self._instance_update_dynamic, instance.processed_devices.copy())
 
-	def instance_unapply_tuning(self, instance, full_rollback = False):
+	def instance_unapply_tuning(self, instance, full_rollback = 1):
 		"""
 		Remove all tunings applied by the plugin instance.
 		"""
+		if full_rollback == 0:
```

Comment:
Will this break profile change? Because we want to make sure that sysctls mentioned in the profile being unloaded are reverted. The new profile might not be touching them.

Reply:
But since it is a parameter, it probably gets decided elsewhere and here it just selects between soft (in-memory) and full rollback, right?

Reply:
> Will this break profile change? 

No

> Because we want to make sure that sysctls mentioned in the profile being unloaded are reverted.

They will be.  During profile change, `stop_tuning()` will be called with "full_rollback == 2".  However, on `TuneD` process termination, it will be called with "full_rollback == 0" if the user specified "rollback = not_on_exit".

> But since it is a parameter, it probably gets decided elsewhere

Yes.